app/src/main/java/com/github/sdp/ratemyepfl/database/ItemsRepository.kt

```diff
@@ -15,32 +15,51 @@ class ItemsRepository @Inject constructor() : ItemsRepositoryInterface, Reposito
     private val collectionRooms = db.collection("rooms")
     private val collectionRestaurants = db.collection("restaurants")
 
-    companion object {
-        private const val TAG_COURSES = "CourseRepository"
-        private const val TAG_ROOMS = "ClassroomRepository"
-        private const val TAG_RESTAURANT = "RestaurantRepository"
-    }
-
-    override suspend fun getCourses(): List<Course?> {
+    override suspend fun getCourses(): List<Course> {
         return getLimit(collectionCourses, 50)
             .mapNotNull { obj -> obj.toCourse() }
     }
 
-    suspend fun getByIdCourses(id: String): Course? {
+    override suspend fun getByIdCourses(id: String): Course? {
         return getById(collectionCourses, id).toCourse()
     }
 
-    override suspend fun getClassrooms(): List<Classroom?> {
+    override suspend fun getClassrooms(): List<Classroom> {
         return getLimit(collectionRooms, 50)
             .mapNotNull { obj -> obj.toClassroom() }
     }
 
-    suspend fun getByIdClassrooms(id: String): Classroom? {
+    override suspend fun getByIdClassrooms(id: String): Classroom? {
         return getById(collectionRooms, id).toClassroom()
     }
 
-    override suspend fun getRestaurants(): List<Restaurant?> {
+    override suspend fun getRestaurants(): List<Restaurant> {
         return getLimit(collectionRestaurants, 50)
             .mapNotNull { obj -> obj.toRestaurant() }
     }
+
+    override suspend fun getByIdRestaurants(id: String): Restaurant? {
+        return getById(collectionRestaurants, id).toRestaurant()
+    }
+
+    // Function to get for a generic Reviewable
+    override suspend fun getById(id : String?): Reviewable? {
+        if(id == null) return null
+        val result = getByIdClassrooms(id)
+        if (result != null) {
+            return result
+        }
+        val result2 = getByIdCourses(id)
+        if (result2 != null) {
+            return result2
+        }
+        return getByIdRestaurants(id)
+    }
+
+    override fun updateRating(rating: ReviewRating, item: Reviewable){
+        val numRatings = item.numRatings + 1
+        val avgRating = (item.avgRating + rating.toValue()) / numRatings
+        db.collection(item.collectionPath).document(item.id).update("numRatings", numRatings.toString(),"avgRating", avgRating.toString())
```

Comment:
Looking at this code and the one in `ReviewsRepository`, I suggest to add a method in `Repository` that match the `getById` structure but only returns the `collection.document(id)`. And call it in methods that apply transformations on the database, like the methods `add()` and `remove()` of `ReviewsRepository` (in which you change `reviewsCollection()` as a private val)

Reply:
I think Alex wants to refactor part of the repositories, I leave this work to him. He could use your ideas @AlexandreMessmer  :).